src/clj/puppetlabs/services/request_handler/request_handler_core.clj

```diff
@@ -12,6 +12,14 @@
             [slingshot.slingshot :as sling]))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Constants
+
+(def error-message-404
+  (str "Error: This master is running Puppet 4, and the URL format has changed. "
+       "This request was likely made with a Puppet 3 Agent, as it used an invalid "
+       "URL format. Please upgrade your agents to Puppet 4 and try again."))
```

Comment:
np.  I think we have about a week or so to come up with the final verbiage.

We may go ahead and merge this PR before we finalize the verbiage, just to keep the machinery rolling, and we can open a follow-up PR to clean it up if that's the case.